### PR TITLE
rename enum Promise to CallTag

### PR DESCRIPTION
**Pull request by BusyJay**
It's confusing when there are both Callback and Promise in enum Promise. So I rename enum Promise to CallTag to make it more clear and straightforward.

### CODE REVIEW

**Comment by andelf** [comment]
how about Promise as a trait

**Comment by BusyJay** [comment]
As I said before, trait/generic can't pass FFI. You can try it yourself. Please let me know if you succeed.

File: src/call/mod.rs [hunk 1]
```diff
@@ -133,14 +133,14 @@ impl Drop for BatchContext {
 fn check_run<F>(bt: BatchType, f: F) -> BatchFuture
     where F: FnOnce(*mut GrpcBatchContext, *mut c_void) -> GrpcCallStatus
 {
-    let (cq_f, prom) = Promise::batch_pair(bt);
-    let prom_box = Box::new(prom);
-    let batch_ptr = prom_box.batch_ctx().unwrap().as_ptr();
-    let prom_ptr = Box::into_raw(prom_box);
-    let code = f(batch_ptr, prom_ptr as *mut c_void);
+    let (cq_f, tag) = CallTag::batch_pair(bt);
+    let tag_box = Box::new(tag);
```
Comment (siddontang) [inline src/call/mod.rs]: can we extract Box::new, batch_ctx, into_raw to a function?

**Comment by disksing** [comment]
LGTM.

**Pull request merged**